htlcswitch/link.go

```diff
@@ -2527,6 +2568,10 @@ func (l *channelLink) updateChannelFee(feePerKw chainfee.SatPerKWeight) error {
 		return err
 	}
 
+	// The fee passed the channel's validation checks, so we update the
+	// mailbox feerate.
+	l.mailBox.SetFeeRate(feePerKw)
```

Comment:
Is it a problem if we set fee rate here but then fail to send the new fee rate through to our peer?

Reply:
The assumption I made here is that `SendMessage` either sends the message or doesn't send the message. If it doesn't send the message, the connection died and the link is shutdown. This assumes that `brontide.Noise` doesn't drop messages or get into some weird state. On restart the mailbox will be called with `SetFeeRate(local commitment fee rate)`.

Typing this out, I realize that the mailbox could have an outdated feerate:
- `---update_fee--->`
- `---commit_sig--->`
- *LHS dies*
- `* AttachMailBox with LocalCommit feerate, which won't get immediately refreshed`

Could add another method to `LightningChannel` to retrieve the remote commitment tip feerate if we're initiator (which will always be the latest), but honestly don't think it's necessary as it should eventually get to the feerate after a link restart or another fee update.

Reply:
comment added